File: files/CSEC 26012025.R
library(tidyverse)
library(readxl)

path = "files/Ex-Challenge 04 2025.xlsx"
input = read_excel(path, range = "B3:D18")
test  = read_excel(path, range = "F3:G13")

result = input %>%
  mutate(Rank = dense_rank(desc(Demand))) %>%
  summarise(`Fruit(s)` = paste0(Fruit, collapse = " ; "), .by = Rank) %>%
  arrange(Rank)

File: files/2025-01-26/CSEC 26012025.py
import pandas as pd

path = "files/Ex-Challenge 04 2025.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=2, nrows=15)
test = pd.read_excel(path, usecols="F:G", skiprows=2, nrows=10)

result = input.assign(Rank=input['Demand'].rank(method='dense', ascending=False).astype(int)) \
              .groupby('Rank')['Fruit'].agg(' ; '.join) \
              .reset_index() \
              .sort_values(by='Rank')

